src/encoding/predictSingletonAddress.ts

```diff
@@ -2,25 +2,47 @@ import { AbiCoder, concat, getCreate2Address, keccak256 } from "ethers";
 import { address as factoryAddress } from "../factory/erc2470Factory";
 import { Create2Args } from "../types";
 
+/**
+ * Predicts the address of a singleton contract deployed using CREATE2.
+ *
+ * @param {Object} params - The function parameters.
+ * @param {string} [params.factory=factoryAddress] - The address of the factory contract.
+ * @param {string} params.bytecode - The bytecode of the contract to deploy.
+ * @param {Object} params.constructorArgs - The constructor arguments for the contract.
+ * @param {any[]} params.constructorArgs.types - The types of the constructor arguments.
+ * @param {any[]} params.constructorArgs.values - The values of the constructor arguments.
+ * @param {BigNumberish} params.salt - The salt value used for CREATE2 deployment.
+ * @returns {string} The predicted address of the singleton contract.
+ */
```

Comment:
Proposal:
/**
 * Predicts the address of a proxy contract deployed via SingletonFactory.
 *
 * Note: The calculation method is the same regardless of whether the factory is Nick or EIP2470.
 *
 * @param {Object} params - The parameters for predicting the address.
 * @param {string} [params.factory=erc3460FactoryAddress] - The address of the factory contract.
 * @param {string} params.bytecode - The bytecode of the contract to deploy.
 * @param {Object} params.constructorArgs - The constructor arguments for the contract.
 * @param {any[]} params.constructorArgs.types - The types of the constructor arguments.
 * @param {any[]} params.constructorArgs.values - Values corresponding to the constructor argument types.
 * @param {string} params.salt - A deployment salt that is internally passed to create2
 *
 * @returns {string} The predicted address of the singleton contract.
 */